docs/runtimes/functions.md

```diff
@@ -1,16 +1,36 @@
-(Functions-container)=
-# Creating and using functions
+(Functions)=
+# Functions
 
+All the executions in MLRun are based on **Serverless Functions**. The functions allow specifying code and 
+all the operational aspects (image, required packages, cpu/mem/gpu resources, storage, environment, etc.). 
+The [different function runtimes](Function_runtimes) take care of automatically transforming the code and spec to fully 
+managed and elastic services over Kubernetes, which saves significant operational overhead, 
+addresses scalability and reduces infrastructure costs.
 
-**In this section**
-```{toctree}
-   :maxdepth: 2
+MLRun supports:
+- Real-time functions for: serving, APIs, and stream processing (based on the high-performance Nuclio engine). 
+- Batch functions (based on Kubernetes jobs, Spark, Dask, Horovod, etc.)
+
+Function objects are all inclusive (code, spec, API, and metadata definitions), which allows placing them 
```

Comment:
need more hyper links in this section, e.g. talk about the fact function resources can be configured and link to the `Managing job resources` page, mention multiple runtimes and link to their doc pages, etc.

Reply:
added